### PR TITLE
Utilize the DbCellValue type

### PR DESCRIPTION
**Pull request by benrr101**
The query/subset API from the tools service now return a DbCellValue[][] as part of the ResultSetSubset. These changes are made to take the DbCellValue and properly determine how to display it in SlickGrid. Contains a vBump for the tools service version.

### CODE REVIEW

File: src/views/htmlcontent/src/js/components/app.component.ts [hunk 1]
```diff
@@ -15,7 +15,8 @@ import { MessagesContextMenu } from './messagescontextmenu.component';
 import {
     IGridIcon,
     IMessage,
-    IRange
+    IRange// ,
+    // DbCellValue
```
Comment (coquagli) [inline src/views/htmlcontent/src/js/components/app.component.ts]: We probably don't need this commented out import, or the comma above.

File: src/views/htmlcontent/src/js/components/app.component.ts [hunk 2]
```diff
@@ -775,4 +779,8 @@ export class AppComponent implements OnInit, AfterViewChecked {
                 }
         });
     }
+
+    static isDbCellValue(object: any): boolean {
```
Comment (coquagli) [inline src/views/htmlcontent/src/js/components/app.component.ts]: I think this belongs in Utils.ts.

File: src/controllers/QueryRunner.ts [hunk 3]
```diff
@@ -281,11 +279,11 @@ export default class QueryRunner {
...
                             let row = result.resultSubset.rows[rowIndex];
-                            let cells = row.slice(range.fromCell, (range.toCell + 1));
-                            if (self.shouldRemoveNewLines()) {
-                                // Remove all new lines from cells
-                                cells = cells.map(x => self.removeNewLines(x));
-                            }
+                            let cellObjects = row.slice(range.fromCell, (range.toCell + 1));
+                            // Remove newlines if requested
+                            let cells = self.shouldRemoveNewLines()
+                                ? cellObjects.map(x => self.removeNewLines(x.displayValue))
```
Comment (coquagli) [inline src/controllers/QueryRunner.ts]: This looks good here. But to verify to others, this now correctly parses an empty array in the case where we are rendering the placeholder values?

File: src/views/htmlcontent/src/js/components/app.component.ts [hunk 4]
```diff
@@ -586,13 +586,16 @@ export class AppComponent implements OnInit, AfterViewChecked {
     /**
      * Format all text to replace all new lines with spaces and performs HTML entity encoding
      */
-    textFormatter(row: number, cell: any, value: string, columnDef: any, dataContext: any): string {
-        let valueToDisplay = value;
+    textFormatter(row: number, cell: any, value: any, columnDef: any, dataContext: any): string {
```
Comment (Raymondd) [inline src/views/htmlcontent/src/js/components/app.component.ts]: You should probably also update the hyperLinkFormatter to handle DBCellValues on line 573 of this file.

File: src/views/htmlcontent/src/js/utils.ts [hunk 5]
```diff
@@ -29,3 +29,12 @@ export function htmlEntities(str: string): string {
...
         : undefined;
 }
+
+/**
+ * Determines if an object is a DbCellValue based on the properties it exposes
+ * @param object The object to check
+ * @returns True if the object is a DbCellValue, false otherwise
+ */
+export function isDbCellValue(object: any): boolean {
+    return (object.displayValue !== undefined && object.isNull !== undefined);
```
Comment (Raymondd) [inline src/views/htmlcontent/src/js/utils.ts]: Might want to also check for object here.
` (object !== undefined && object.displayValue !== undefined && object.isNull !== undefined);`

Comment (benrr101) [inline src/views/htmlcontent/src/js/utils.ts]: Good catch

File: src/views/htmlcontent/src/js/components/app.component.ts (hunk 4 above)
Comment (benrr101) [inline src/views/htmlcontent/src/js/components/app.component.ts]: The hyperlink formatter should be corrected as per your request. Interestingly, I didn't find any unit tests that check the function of the hyperlink formatter

**Pull request merged**